packages/react-component-library/src/hooks/useExternalId.ts

```diff
@@ -0,0 +1,11 @@
+import { useMemo } from 'react'
+import { v4 as uuidv4 } from 'uuid'
+
+/**
+ * Return a memoised unique ID as a string.
+ *
+ * Returns externalId if provided, otherwise generates a random ID.
+ */
+export function useExternalId(externalId?: string) {
```

Comment:
Note: [A `useId` is being added in React 18](https://github.com/reactwg/react-18/discussions/111), so I avoided that name.